tests/lambda_endpoints/test_spice_indexer_lambda.py

```diff
@@ -0,0 +1,141 @@
+
+
+def put_local_file_in_bucket(s3_client, path_in_s3, path_local):
+    """Put the a local file into a test bucket, and return a mock event notification.
+
+    Parameters
+    ----------
+    s3_client
+        The s3 client to use
+    path_in_s3: str
+        The path to place the file in S3
+    path_local: str
+        The local path of the file to upload
+
+    Returns
+    -------
+    s3_event: dict
+        A dictionary mocking the s3 put event notification
+
+    """
+    with open(path_local, "rb") as f:
+        s3_client.put_object(
+            Bucket="test-data-bucket",
+            Key=path_in_s3,
+            Body=f,
+        )
+    event = {
+        "detail-type": "Object Created",
+        "source": "aws.s3",
+        "time": datetime.now().isoformat(),
+        "detail": {
+            "version": "0",
+            "bucket": {"name": "test-data-bucket"},
+            "object": {
+                "key": (path_in_s3),
+                "reason": "PutObject",
+            },
+        },
+    }
+    return event
+
+
+def test_s3_spice_files(session, s3_client, events_client):
+    """Test s3 event.
+
+    The following test mimics a leapsecond kernel being placed on the SDS,
+    followed by a spacecraft clock kernel, and then an attitude file.
+
+    The files are located in the "test_spice_files" directory.
+
+    """
+    current_path = os.path.dirname(os.path.abspath(__file__))
+
+    # Insert leapsecond spice kernel
+    leapsecond_event = put_local_file_in_bucket(
+        s3_client,
+        "spice/lsk/naif0012.tls",
+        current_path + "/test_spice_files/naif0012.tls",
+    )
+    spice_indexer.lambda_handler(leapsecond_event, None)
+
+    # Insert spacecraft clock spice kernel
+    clock_kernel_event = put_local_file_in_bucket(
+        s3_client,
+        "spice/sclk/imapsclk_0012.tsc",
+        current_path + "/test_spice_files/imapsclk_0012.tsc",
+    )
+    spice_indexer.lambda_handler(clock_kernel_event, None)
+
+    # Insert a new attitude kernel
+    attitude_kernel_event = put_local_file_in_bucket(
+        s3_client,
+        "spice/ck/imap_2025_118_2025_120_001.ah.bc",
+        current_path + "/test_spice_files/imap_2025_118_2025_120_001.ah.bc",
+    )
+    spice_indexer.lambda_handler(attitude_kernel_event, None)
+
+    # Verify that the file was moved to the /tmp directory
+    # (conftest.py sets the SPICE directory as "tmp")
+    assert os.path.exists("/tmp/lsk/naif0012.tls")
+    assert os.path.exists("/tmp/sclk/imapsclk_0012.tsc")
+    assert os.path.exists("/tmp/ck/imap_2025_118_2025_120_001.ah.bc")
+
+    # Verify that the database was populated appropriately
+    result = session.query(models.SPICEFiles).all()
+
+    # Verify the database has 3 files:
+    assert len(result) == 3
+
+    # Loop through the 3 database entries and ensure their accuracy
+    leapsecond_kernel_reached = False
+    clock_kernel_reached = False
+    attitude_kernel_reached = False
+    for r in result:
```

Comment:
Rather than looping through, we should try to query for an individual file and assert on that. Do 3 separate queries here on your expectations.

Reply:
True, I was also thinking this would help show that no more than those 3 were inserted, but I guess we don't need to test that